Reintroduce Activity Logs callout in Dashboard v2

andrea-sdl opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Ref: DOTDASH-432

## Proposed Changes

* Implement the callout for Activity Log
* Change the "No results" message for Activity Log into "No activities" (when there is 0 entries) and "No activity found" when the search result leads to no result.
* Switch the PHP/Web Server logs callout to not being an overlay

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Activity Log: We want to reintroduce the callout to allow users to upgrade their plan, the callout is a little bit difference since it relies on having a gradient that lets the last item disappear a little bit.
* PHP/Web Server logs: The previous callout, with the overlay would cover the entire tabs, we wanted the overlay to be present but still allow the tabs to be usable since the Activity Log is available (albeit limited) on the free version. 

One thing to note is that, for the Activity Logs callout, I used the `ActivityLogsCallout` instead of passing through the `HostingFeatureGatedWithCallout` component since, in this case, we want to print it "Alongside" the extra content, and using the `HostingFeatureGatedWithCallout` with an empty content didn't feel very consistent.

| Scenario | Before | After |
|----------|--------|------|
| Free plan with no activities | <img width="2338" height="1980" alt="CleanShot 2025-09-22 at 12 45 40@2x" src="https://github.com/user-attachments/assets/8caa20a0-d4c2-47e3-ba3e-ed382c319511" /> | <img width="2814" height="1742" alt="CleanShot 2025-09-22 at 15 39 51@2x" src="https://github.com/user-attachments/assets/2b528a5f-0f0a-47b7-95bc-aca30128b3f7" /> |
| Free plan with activities | <img width="1952" height="1734" alt="CleanShot 2025-09-22 at 15 41 40@2x" src="https://github.com/user-attachments/assets/47203d84-15a6-4bf4-8edc-b41f330e6a8f" /> | <img width="2570" height="2216" alt="CleanShot 2025-09-22 at 15 43 45@2x" src="https://github.com/user-attachments/assets/a29cf36c-3c7f-490c-84c5-bfa68de5c27f" /> |
| Gated PHP/WebServer logs | <img width="3036" height="1430" alt="CleanShot 2025-09-22 at 16 11 52@2x" src="https://github.com/user-attachments/assets/8e149df0-fb5d-41fb-9405-7a5a52de735b" /> | <img width="2598" height="1386" alt="CleanShot 2025-09-22 at 16 11 30@2x" src="https://github.com/user-attachments/assets/33c20e88-e2b2-4d6c-98b1-b4e7b1fcda99" /> |


## Testing Instructions

To test this, I suggest having
1. One new site on the free plan (with no activity)
2. One site on the free plan with some activity
3. One site on a paid plan with some activity

**Free site with no activity**
* `http://calypso.localhost:3000/v2/sites/<site>/logs/activity` Test that you see the callout
* Click on the "PHP" tabs, ensure you see the callout and that the tabs are still navigable.

**Free site with activity**
* `http://calypso.localhost:3000/v2/sites/<site>/logs/activity` Test that you see the callout and that the last item is slightly hidden by the gradient
* Click on the "PHP" tabs, ensure you see the callout and that the tabs are still navigable.

**Paid site with activity**
* `http://calypso.localhost:3000/v2/sites/<site>/logs/activity`  No callout should be present, all should be working.
* Click on the "PHP" tabs, ensure there is no callout.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

andrea-sdl committed:
Add support for activity log callout and move the logs callout inside the tab.

andrea-sdl committed:
adjust callout padding

andrea-sdl committed:
add better no results message for activity log

andrea-sdl committed:
improve bottom padding for activity log callout

andrea-sdl committed:
refactor: enhance type safety for activity logs and improve access check

andrea-sdl committed:
refactor: rename access prop for clarity in activity logs component

andrea-sdl committed:
refactor: improve structure and styling for activity logs component

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

fushar commented on client/dashboard/sites/logs/index.tsx:
```diff
@@ -62,74 +70,87 @@ function SiteLogs( { logType }: { logType: LogType } ) {
 	};
-
+	const hasActivityLogAccess =
+		hasHostingFeature( site, HostingFeatures.ACTIVITY_LOG ) ||
```
I believe, we can drop `hasHostingFeature()` from the conditionals here here as it's subset of `hasPlanFeature()` (but please do double check)

andrea-sdl replied:
`hasHostingFeature` returns false if the plan has expired, I think it's probably better.

fushar replied:
Interesting, but doesn't that mean expired plans still have access to the feature? Because `hasPlanFeature()` will return true (we're using `||` here).

I think what we probably want here is `hasPlanFeature() && ! site.plan?.expired`.

fushar commented on client/dashboard/sites/logs/style.scss:
```diff
@@ -2,3 +2,29 @@
+	&__server,
+	&__php {
+		.dashboard-callout {
+			max-width: 889px;
```
Should this be 600px so that it's the same as the previous callout width 🤔

andrea-sdl replied:
Yeah, I kept the two callouts with the same width for consistency (this way the callout on the Activity tab will be the same size as the other's). We can change it to 600px if we feel it's better.

fushar replied:
OK Let's keep the original design here then.

andrea-sdl replied:
I also pinged design in p1758638332676449/1758295409.699779-slack-C09B6MEPCPN to get their take on the new callouts as well as the empty state. I'm wondering if we should aim to have something similar to what we have for the Scans page, where we use the `DataViewsEmptyState` component to show a nicer empty status.

The only thing to keep in mind is that, on the free plan, the page then becomes suddenly busy (see this example I created to show the concept)
<img width="720" height="601" alt="image" src="https://github.com/user-attachments/assets/771c409f-31af-4c42-bbca-ed2d516c6ec8" />

fushar commented on client/dashboard/sites/logs/style.scss:
```diff
@@ -2,3 +2,29 @@
+			z-index: 2;
+
+			&-content {
+				max-width: 889px;
```
Let's check if we can remove some of the CSS here. For example, I don't think this width is correct. If I remove it, I get the correct width based on Figma

|This PR|If this is removed|
|-|-|
|<img width="912" height="276" alt="Image" src="https://github.com/user-attachments/assets/f176bb3e-83a3-47fa-90c5-e34eeb4ce17b" />|<img width="695" height="301" alt="Image" src="https://github.com/user-attachments/assets/48a710dd-e846-4acb-9cb9-83549a768f3c" />

fushar replied:
Oh my god... I think I am looking the wrong thing. The image in the linked issue is different from the actual Figma 🤦 

<img width="659" height="310" alt="image" src="https://github.com/user-attachments/assets/8283839d-2e21-4e42-a4a6-c506d284d708" />

fushar replied:
I am now becoming more confused as the Figma contains two callouts of the same thing but with different aspect ratio 😄 that's why I thought it's smaller 🤔 

<img width="962" height="260" alt="image" src="https://github.com/user-attachments/assets/fdf946bf-980d-410b-8b14-d24b58c1b4a3" />

keoshi replied:
Thanks for noticing this, @fushar! Might be our bad in how we structured the component. The left banner has less copy, so it will be shorter, and the image will start to crop.

I've asked Dotcom Design if we have any guidelines on how to treat these illustrations here: QOBjujZah0UlGDDdLKZhzi-fi-4204_245056#1438375692

fushar commented on client/dashboard/sites/logs-activity/activity-logs-callout-illustration.svg:
```diff
@@ -0,0 +1,54 @@
+<svg width="399" height="204" viewBox="0 0 399 204" fill="none"
```
I think this image is a bit cropped? The correct one should be:

|This PR|Correct based on Figma|
|-|-|
|<img width="295" height="248" alt="Image" src="https://github.com/user-attachments/assets/698b17aa-d117-4585-bd93-14d73de8c3c3" />|<img width="195" height="129" alt="Image" src="https://github.com/user-attachments/assets/74b2c9e2-b8f3-4bca-ab92-9c7eebd69fa6" />

fushar replied:
Similarly, I was looking at the wrong thing... but I wonder why the image on Figma is "cropped" like this 😅

andrea-sdl replied:
You mean why it's rectangular?

fushar replied:
I mean why the corner circles are cropped.

andrea-sdl replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/105930/commits/f28be0fc61ccf34833c6be9867731939d42c0375 , I used the wrong source for the SVG and the square surrounding the circles was cut.

fushar reviewed (commented):
I think generally this PR is in a good direction. I don't have a better alternative. Maybe @arthur791004 wants to take a look since he's been working on a related refactor 🤔

fushar reviewed (commented):
I think the overlay has too much margin from the content? So the blur effect is not that visible 🤔 

<img width="1257" height="643" alt="Image" src="https://github.com/user-attachments/assets/26ff6515-607a-4741-90e9-863fd71a7421" />

andrea-sdl committed:
make callout width 886px

andrea-sdl committed:
add missing callout text

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

andrea-sdl committed:
simplify CSS to reuse site-logs-card css

andrea-sdl committed:
fix wrong svg (too much cropping)

andrea-sdl committed:
change padding/margin to let the gradient start earlier

keoshi reviewed (commented):
In general this looks really good to me. Thanks for going this, @andrea-sdl!

The callout looks great and the gradient looks beautiful. As do the other two pages.

<img width="1391" height="685" alt="image" src="https://github.com/user-attachments/assets/5fc92840-de81-4625-8705-b6a1e918e1a0" />

There's an minor issue with the [illustration](https://github.com/Automattic/wp-calypso/pull/105930#discussion_r2372715060), that we are discussing here. Other than that, we could tweak the copy to underline the free plan can only see the latest 20 events:

> Debug issues faster with insights from a comprehensive audit log of all your admin activities.
> 
> With your free plan, you can see your 20 most recent events. Upgrade for 30 days of history, plus filtering and date range controls. Available on the WordPress.com Business and Commerce plans.

---

Adding a few nitpicky notes here, but if outside of the scope of this PR (which most seem to be) feel free to disregard.

#### Gradient is capturing the cursor

With the gradient in place, it's impossible to hover over a row or press the ellipsis menu. We should probably add a `pointer-events: none` or similar to the gradient.

#### Tabs element seems space-y at the top

Feels a bit unbalanced with extra space on top. Ideally, “Activity” would sit in the middle of the two horizontal lines. I've also commented this on a PR for Scan, so I'm assuming it's a global thing, probably at the component level. CC'ing @Initsogar in case there's a shared solution.

<img width="566" height="105" alt="image" src="https://github.com/user-attachments/assets/2871d8f5-267c-4d09-9f84-f6480bc1ea34" />

#### Lack of search makes the header seem empty

Not sure why the search doesn't show up, but not having it there leaves a lot of empty space. Makes it seem the top section is somewhat buggy.

<img width="1383" height="124" alt="image" src="https://github.com/user-attachments/assets/d58236d0-8d32-4869-8d24-4f752e9a2f13" />

andrea-sdl committed:
edit callout message

andrea-sdl committed:
ignore pointer events from the callout gradient but not from the callout itself

andrea-sdl committed:
hide settings and search for free users

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

fushar commented on client/dashboard/sites/logs-activity/dataviews/index.tsx:
```diff
@@ -63,9 +71,12 @@ function SiteActivityLogsDataViews( {
 			fields={ fields as Field< SiteActivityLog >[] }
 			view={ view }
 			actions={ actions }
+			config={ hasActivityLogsAccess ? undefined : { perPageSizes: [ 20 ] } } // Disable changing perPage if no access
```
Should this be 


```suggestion
			config={ hasActivityLogsAccess ? undefined : { perPageSizes: [ ACTIVITY_LOGS_DEFAULT_PAGE_SIZE ] } } // Disable changing perPage if no access
```

andrea-sdl replied:
Good catch! Added in 8a50f236f0b55a80d8b706ac7cf922e006b803a1 (after I resolved the conflicts with trunk)

fushar reviewed (approved):
Tested and it works well!

It's a bit unfortunate to have too many CSS here. But due to the nature of the design, I can't think of better way. I think it's fine for now

fushar commented on client/dashboard/sites/logs-activity/dataviews/index.tsx:
```diff
@@ -63,9 +71,12 @@ function SiteActivityLogsDataViews( {
 			defaultLayouts={ { table: {} } }
 			onChangeView={ onChangeView }
+			empty={ <p>{ view.search ? __( 'No activity found' ) : __( 'No activities' ) }</p> }
+			children={ hasActivityLogsAccess ? undefined : <DataViews.Layout /> } // showing only the layout when on the free plan.
```
Actually I don't quite get this -- why do we want to show the controls on Free plan?

andrea-sdl replied:
> why do we want to show the controls on Free plan?

@fushar This is actually doing the opposite. It's showing only the Layout _without_ the controls, just like we currently have on v1.

fushar replied:
Oops, I misread the comments 😂

andrea-sdl replied:
Here are some screens to compare the experience 
<img width="2926" height="1394" alt="CleanShot 2025-09-29 at 11 53 35@2x" src="https://github.com/user-attachments/assets/af4be817-b279-4cfc-b9bd-9099808bfb1f" />
<img width="2844" height="1010" alt="CleanShot 2025-09-29 at 11 54 08@2x" src="https://github.com/user-attachments/assets/abaaafd7-7fa0-4a31-8b68-e3a4968429d5" />

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

andrea-sdl committed:
use ACTIVITY_LOGS_DEFAULT_PAGE_SIZE constant instead of hardcoding the value

andrea-sdl committed:
fix test typing

arthur791004 commented on client/dashboard/sites/logs/style.scss:
```diff
@@ -1,6 +1,33 @@
+	&--server,
+	&--php {
+		.dashboard-callout {
+			max-width: 886px;
```
It seems better to keep the `max-width: 600px` as it doesn't look good 🥲

andrea-sdl replied:
I went with 886 to have consistency in how it's displayed across the 3 tabs since the Activity Callout is wider.

I have no strong opinions on this, so I'm going with 600px if you think it's better. Here is the comparison. 

<img width="2862" height="1510" alt="CleanShot 2025-09-29 at 12 20 58@2x" src="https://github.com/user-attachments/assets/e00808fa-11e8-44a6-94d9-bdd9fc20e1b7" />
<img width="2942" height="1366" alt="CleanShot 2025-09-29 at 12 21 19@2x" src="https://github.com/user-attachments/assets/d131a9fc-b46e-4cb5-81bf-9b6f7931dc1b" />

andrea-sdl committed:
Hide date range picker based on activity log access and log type

andrea-sdl commented:
@arthur791004 Thanks for the feedback, I'm going through it.

> It looks weird that we have the DatePicker on the screen even if the feature is not available 😂

fixed in 3f79ff99eeb228a777b9310e901d1dc13b8a3348

andrea-sdl committed:
Make the PHP/Web Server logs callout smaller

andrea-sdl committed:
fix callout on mobile not having enough padding

andrea-sdl commented:
> We have to adapt the mobile view as well 🙂

Thanks @arthur791004 ! Addressed it in https://github.com/Automattic/wp-calypso/pull/105930/commits/9cd7a33095348e7adf4e2e7c7d7860d93a523168

Here's how it looks, let me know if you think we need some extra adjustements (the longer text is making the callout be quite high in height, sadly)
<img width="644" height="1274" alt="CleanShot 2025-09-29 at 12 31 11@2x" src="https://github.com/user-attachments/assets/91db2b68-7023-4c81-b907-fdfe94251ca5" />

fushar commented on client/dashboard/sites/logs/index.tsx:
```diff
@@ -90,19 +93,20 @@ function SiteLogs( { logType }: { logType: LogType } ) {
+		hasHostingFeature( site, HostingFeatures.ACTIVITY_LOG ) ||
+		hasPlanFeature( site, HostingFeatures.ACTIVITY_LOG );
+	// hide the datepicker if the user doesn't have access to activity logs or doesn't have logging feature at all
+	const shouldShowDateRangePicker =
```
CMIIW but I think this is simply

```suggestion
	const shouldShowDateRangePicker = hasHostingFeature( site, HostingFeatures.LOGS );
```

andrea-sdl replied:
I think you're correct here, although I wonder if there are some edge cases I might miss. After some testing, though, it seems to me that if we don't have logs, at least on WordPress.com, then we won't also have Activity Logs.

andrea-sdl committed:
Update client/dashboard/sites/logs/index.tsx

Co-authored-by: Ashar Fuadi <ashar.fuadi@automattic.com>

andrea-sdl committed:
simplify conditional

andrea-sdl committed:
fix callout on mobile not having enough padding

andrea-sdl commented:
> It still looks not good 😢 But we can address it in the follow-up PR.

I think I missed the error because Chrome was caching some live CSS changes I did while testing things out. Fixed (hopefully :D ) on https://github.com/Automattic/wp-calypso/pull/105930/commits/76b614e3d8591e331d52eddf44125debc2b043b1

Sorry for the back and forth here, I was 100% sure I already fixed this, so thank you for the patience! @arthur791004

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

pandah3 reviewed (commented):
Thanks for iterating on this!!

I'm getting a flash of the callout before seeing the logs (free plan with logs):

https://github.com/user-attachments/assets/0b5528cf-0253-4d46-a71a-d8505bc4dcdf

Minor:

I also see the callout shift when I come back to it from a different tab. Not sure if it's worth spending lots of time to fix this, but thought I'd mention it.

https://github.com/user-attachments/assets/ac742991-0481-4608-8e50-06f61b417426

arthur791004 commented:
> I'm getting a flash of the callout before seeing the logs (free plan with logs):
> I also see the callout shift when I come back to it from a different tab. Not sure if it's worth spending lots of time to fix this, but thought I'd mention it.

Both are due to the loading state...😓

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

andrea-sdl committed:
move the activity log callout inside the dataview to make it aware of the loading state and avoid the callout moving up/down while the loading finishes

andrea-sdl committed:
add some animation to the callout so that it doesn't appear out of nowhere but feels like a continuation of the screen

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

pandah3 reviewed (approved):
No more flashing for me, woohoo!

The video you included won't play for some reason, but what I see lines up with the comment you wrote about the change, so I think it's as expected! Thanks for addressing!!

https://github.com/user-attachments/assets/02372248-b335-4024-9464-84cf0de83f5d

andrea-sdl committed:
Merge branch 'trunk' into add/activity-logs-v2-upsell-DOTDASH-432

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/20324432

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @andrea-sdl for including a screenshot in the description! This is really helpful for our translators.
